### DISCUSSION TITLE
Should I adopt Memory engine for big data if I have sufficient memory

### DISCUSSION THREAD

**Discussion by Alex-Cheng** [ClickHouse/ClickHouse#34675] [Q&A] [answered]
I have the server with sufficient server and I need to do real-time analysis on a data of 20G. Should I adopt memory engine instead of MergeTree engine?

**Comment by qoega** [comment, marked as answer]
Most likely you should not. MergeTree will have your data sorted and some more optimizations will be applied.
You can easily try both just creating one table from another and check your queries. And you may want to add indices or codecs for more efficient queries

**Reply by Alex-Cheng** [reply]
Thank you. However, my experiment result shows that the memory engine is 3 times faster than MergeTree engine. Maybe I did not leverage the capability of MergeTree engine very well.

**Reply by Alex-Cheng** [reply]
How can I avoid the disk I/O when I adopt Merge Tree engine? If my memory is big enough, will it put the data into memory?

**Reply by Alex-Cheng** [reply]
why my experiment shows that memory engine is faster than MergeTree engine? Is it because I misused MergeTree engine?
